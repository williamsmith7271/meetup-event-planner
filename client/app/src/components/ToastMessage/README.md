## ToastMessage Component
A toast component that shows up at the top of the screen with an alert.
message,
onClose,
status,
### Example

```js
<ToastMessage />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **message**    | String   |             | Any string value
| **onClose**    | Func   |             | A callback when requesting to close
| **status**    | String / Enum   |  'ok'           | An enum value, one of: critical|warning|ok|disabled|unknown


### Other Information
See: https://grommet.github.io/docs/toast
