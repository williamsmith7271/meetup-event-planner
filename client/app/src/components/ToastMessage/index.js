import React, { PropTypes } from 'react';
import Toast from 'grommet-udacity/components/Toast';

const ToastMessage = ({
  message,
  onClose,
  status,
}) => (
  <Toast status={status} onClose={onClose}>
    {message}
  </Toast>
);

ToastMessage.propTypes = {
  message: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  status: PropTypes.oneOf([
    'critical',
    'warning',
    'ok',
    'disabled',
    'unknown',
  ]),
};

ToastMessage.defaultProps = {
  status: 'ok',
};

export default ToastMessage;
