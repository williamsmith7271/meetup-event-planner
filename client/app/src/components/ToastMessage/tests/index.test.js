
import ToastMessage from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<ToastMessage />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <ToastMessage message="Thanks!" onClose={e => e} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
