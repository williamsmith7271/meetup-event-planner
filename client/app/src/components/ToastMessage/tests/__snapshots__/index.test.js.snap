exports[`<ToastMessage /> should render with default props 1`] = `
<Toast
  onClose={[Function onClose]}
  status="ok">
  Thanks!
</Toast>
`;
